internal/services/managedredis/managed_redis_access_policy_assignment_resource.go

```diff
@@ -0,0 +1,172 @@
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IsUUID,
+		},
+	}
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) ModelObject() interface{} {
+	return &ManagedRedisAccessPolicyAssignmentResourceModel{}
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) ResourceType() string {
+	return "azurerm_managed_redis_access_policy_assignment"
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return databases.ValidateAccessPolicyAssignmentID
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ManagedRedisAccessPolicyAssignmentResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.ManagedRedis.DatabaseClient
+
+			clusterId, err := redisenterprise.ParseRedisEnterpriseID(model.ManagedRedisID)
+			if err != nil {
+				return err
+			}
+
+			// Access policy assignments are created on the specified database
+			// Use object_id as the assignment name to ensure one assignment per user per database
+			dbId := databases.NewDatabaseID(clusterId.SubscriptionId, clusterId.ResourceGroupName, clusterId.RedisEnterpriseName, defaultDatabaseName)
+			id := databases.NewAccessPolicyAssignmentID(clusterId.SubscriptionId, clusterId.ResourceGroupName, clusterId.RedisEnterpriseName, defaultDatabaseName, model.ObjectID)
+
+			existing, err := client.AccessPolicyAssignmentGet(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			dbResp, err := client.Get(ctx, dbId)
+			if err != nil {
+				if response.WasNotFound(dbResp.HttpResponse) {
+					return fmt.Errorf("managed Redis database %s was not found", dbId)
+				}
+				return fmt.Errorf("retrieving %s: %+v", dbId, err)
+			}
+
+			createInput := databases.AccessPolicyAssignment{
+				Name: pointer.To(model.ObjectID),
+				Properties: &databases.AccessPolicyAssignmentProperties{
+					AccessPolicyName: "default",
+					User: databases.AccessPolicyAssignmentPropertiesUser{
+						ObjectId: pointer.To(model.ObjectID),
+					},
+				},
+			}
+
+			if err := client.AccessPolicyAssignmentCreateUpdateThenPoll(ctx, id, createInput); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
```

Comment:
For read, the default timeout is 5 mins.
```suggestion
		Timeout: 5 * time.Minute,
```